pkg/store/cache/inmemory.go

```diff
@@ -94,11 +112,27 @@ func NewInMemoryIndexCacheWithConfig(logger log.Logger, commonMetrics *commonMet
 	}
 
 	c := &InMemoryIndexCache{
+		commonMetrics:    commonMetrics,
 		logger:           logger,
 		maxSizeBytes:     uint64(config.MaxSize),
 		maxItemSizeBytes: uint64(config.MaxItemSize),
-		commonMetrics:    commonMetrics,
 	}
+	cache, err := ristretto.NewCache(&ristretto.Config{
+		NumCounters: int64(config.MaxSize) / 1024,
```

Comment:
We probably need a better way to determine this. `NumCounters` should be set to expected items in the cache * 10